wpilibj/src/main/java/edu/wpi/first/wpilibj/IterativeRobotBase.java

```diff
@@ -280,6 +284,9 @@ protected void loopFunc() {
       HAL.observeUserProgramTest();
       testPeriodic();
       m_watchdog.addEpoch("testPeriodic()");
+    } else {
```

Comment:
This isn't necessary with the other change.

Reply:
It's an extra precaution. Does it harm anyone?

Reply:
I think we should go for as simple as possible here with as few checks as possible. You could do this:

```
if isDisabled()
else if isAutonomous()
else if !isTest() // in teleop
else // in test
```
This would also allow you to remove the other method. I don't think you should have code that will never execute unless something is wrong with WPILib itself. If something is THAT wrong, then an exception can be thrown. But, with the code I suggested, it makes doing that pointless.